Mathlib/Algebra/Ring/Ext.lean

```diff
@@ -0,0 +1,528 @@
+/-
+Copyright (c) 2024 Raghuram Sundararajan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Raghuram Sundararajan
+-/
+import Mathlib.Algebra.Ring.Defs
+import Mathlib.Algebra.Group.Ext
+-- import Mathlib
+
+/-!
+# Extensionality lemmas for rings and similar structures
+
+In this file we prove extensionality lemmas for the ring-like structures defined in
+`Mathlib/Algebra/Ring/Defs.lean`, ranging from `NonUnitalNonAssocSemiring` to `CommRing`. These
+extensionality lemmas take the form of asserting that two algebraic structures on a type are equal
+whenever the addition and multiplication defined by them are both the same.
+
+## Implementation details
+
+We follow `Mathlib/Algebra/Group/Ext.lean` in using the term `(letI := i; HMul.hMul : R → R → R)` to
+refer to the multiplication specified by a typeclass instance `i` on a type `R` (and similarly for
+addition). We abbreviate these using some local notations.
+
+Since `Mathlib/Algebra/Group/Ext.lean` proved several injectivity lemmas, we do so as well — even if
+sometimes we don't need them to prove extensionality.
+
+## Tags
+semiring, ring, extensionality
+-/
+
+local macro:65 lhs:term:65 "+[" type:term ", " inst:term "]" rhs:term:66 : term =>
+  `(term| (letI := $inst; ($lhs : $type) + ($rhs : $type) : $type))
+local macro:70 lhs:term:70 "*[" type:term ", " inst:term "]" rhs:term:71 : term =>
+  `(term| (letI := $inst; ($lhs : $type) * ($rhs : $type) : $type))
+
+universe u
+
+variable {R : Type u}
+
+/-! ### Distrib -/
+namespace Distrib
+
+@[ext] theorem ext ⦃inst₁ inst₂ : Distrib R⦄
+    (h_add : ∀ x y, x +[R, inst₁] y = x +[R, inst₂] y)
+    (h_mul : ∀ x y, x *[R, inst₁] y = x *[R, inst₂] y) :
+    inst₁ = inst₂ := by
+  -- Split into `add` and `mul` functions and properties.
+  rcases inst₁ with @⟨⟨⟩, ⟨⟩⟩
+  rcases inst₂ with @⟨⟨⟩, ⟨⟩⟩
+  -- Prove equality of parts using function extensionality.
+  congr <;> (apply funext₂; assumption)
+
+theorem ext_iff (inst₁ inst₂ : Distrib R) :
+    inst₁ = inst₂ ↔
+      (∀ x y, x +[R, inst₁] y = x +[R, inst₂] y)
+      ∧ (∀ x y, x *[R, inst₁] y = x *[R, inst₂] y) :=
```

Comment:
Nitpick: we put operators before a linebreak. Could you also make the same change below?
```suggestion
      (∀ x y, x +[R, inst₁] y = x +[R, inst₂] y) ∧
        (∀ x y, x *[R, inst₁] y = x *[R, inst₂] y) :=
```

Reply:
Done in 6580ed52468b94fd5c04718c6bde7b45564511d5.